src/datetime/tests.rs

```diff
@@ -639,26 +632,10 @@ fn test_datetime_format_with_local() {
 }
 
 #[test]
-#[cfg(feature = "clock")]
-fn test_datetime_is_copy() {
-    // UTC is known to be `Copy`.
-    let a = Utc::now();
-    let b = a;
-    assert_eq!(a, b);
-}
-
-#[test]
-#[cfg(feature = "clock")]
-fn test_datetime_is_send() {
-    use std::thread;
-
-    // UTC is known to be `Send`.
-    let a = Utc::now();
-    thread::spawn(move || {
-        let _ = a;
-    })
-    .join()
-    .unwrap();
+fn test_datetime_is_send_and_copy() {
```

Comment:
hmm... this is beyond the edge of my rust knowledge.
Does this test `test_datetime_is_send_and_copy` invoke trait `Send` on a `Utc`? And trait `Copy`?
I can see function `_assert_send_copy` requires `T` to have those traits, but AFAICT, those traits are not exercised here.

Reply:
Learned it here https://github.com/rust-lang/rust/pull/30872#issuecomment-173014586.

Reply:
I created a little example of the old code here that tested the `Copy` trait. Suprisingly, I couldn't get the underlying `clone()` to print anything (I'm pretty sure `clone()` is invoked by doing a copy) so I'm not sure what's happening.

```rust
fn main() {
    println!("Hello, world!");
}

#[derive(Debug, PartialEq, Eq)]
pub struct Object2 {}
impl Object2 {
    pub fn new() -> Self {
        Object2 {}
    }
}
impl Clone for Object2 {
    fn clone(&self) -> Object2 {
        eprintln!("Clone");
        *self
    }
}
impl Copy for Object2 {}

#[test]
fn test_copy() {
    eprintln!("test_copy start...");
    let a = Object2::new();
    let b = a;
    assert_eq!(a, b);
}
```

Run with `cargo test -v -- --nocapture`

I guess your change is probably okay... 🤔

Reply:
It works the other way around. `Clone` may use a simple `memcpy`, but `Copy` is guaranteed to be just a copy andnot call any functions.